<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8"/>
<title>Snakemake Workshop</title>
<meta name="author" content="(Edmund Miller)"/>
<style type="text/css">
.underline { text-decoration: underline; }
</style>
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/reveal.js/css/reveal.css"/>

<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/reveal.js/css/theme/white.css" id="theme"/>


<!-- If the query includes 'print-pdf', include the PDF print sheet -->
<script>
    if( window.location.search.match( /print-pdf/gi ) ) {
        var link = document.createElement( 'link' );
        link.rel = 'stylesheet';
        link.type = 'text/css';
        link.href = 'https://cdn.jsdelivr.net/npm/reveal.js/css/print/pdf.css';
        document.getElementsByTagName( 'head' )[0].appendChild( link );
    }
</script>
</head>
<body>
<div class="reveal">
<div class="slides">
<section>
<section id="slide-orgbcb7d0b">
<h2 id="orgbcb7d0b">Snakemake</h2>
</section>
</section>
<section>
<section id="slide-orgbf2fb14">
<h2 id="orgbf2fb14">Setting up</h2>
<div class="outline-text-2" id="text-orgbf2fb14">
</div>
</section>
</section>
<section>
<section id="slide-org1c287ba">
<h3 id="org1c287ba">Download Vscode</h3>
<div class="outline-text-3" id="text-org1c287ba">
</div>
</section>
<section id="slide-org7a055e7">
<h4 id="org7a055e7">Extensions</h4>
<ul>
<li><a href="https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-ssh">Remote extension</a></li>

</ul>
<p>
On Remote:
</p>
<ul>
<li><a href="https://marketplace.visualstudio.com/items?itemName=alping.vscode-snakemake">Snakemake extension</a></li>

</ul>
</section>
</section>
<section>
<section id="slide-orgf5efc37">
<h3 id="orgf5efc37">Logging into promoter</h3>
<div class="outline-text-3" id="text-orgf5efc37">
</div>
</section>
<section id="slide-org6309779">
<h4 id="org6309779">Using VSCode Remote extension&#xa0;&#xa0;&#xa0;<span class="tag"><span class="suhana">suhana</span></span></h4>
<p>
NOTE: YOU MUST BE ON <code>COMETNET</code>
</p>
<div class="org-src-container">

<pre  class="src src-shell"><code trim>test@promoter
test123
</code></pre>
</div>
</section>
</section>
<section>
<section id="slide-org1399d99">
<h3 id="org1399d99">Creating a new workflow with <code>cookiecutter</code></h3>
<div class="outline-text-3" id="text-org1399d99">
</div>
</section>
<section id="slide-org0f338d0">
<h4 id="org0f338d0">Open a terminal</h4>
<ul>
<li><code>Ctrl + Shift + `</code></li>
<li>Use &lt;your-name&gt; for <code>repo_name</code></li>

</ul>
<div class="org-src-container">

<pre  class="src src-shell"><code trim>cookiecutter gl:functional-genomics/cookiecutter-snakemake-workflow
<span style="color: #ECBE7B;">cd</span> &lt;your_name&gt;
<span style="color: #ECBE7B;">make</span> install
</code></pre>
</div>
</section>
<section id="slide-org17b76c6">
<h4 id="org17b76c6">Running <code>snakemake</code> for the first time</h4>
<div class="org-src-container">

<pre  class="src src-shell"><code trim><span style="color: #ECBE7B;">make</span> dry
<span style="color: #5B6268;"># </span><span style="color: #5B6268;">Equivalent to</span>
poetry run snakemake -n --use-conda --use-singularity -j <span style="color: #da8548; font-weight: bold;">4</span> --report report.html
</code></pre>
</div>
<ul>
<li>Right click on <code>report.html</code> and download it locally</li>
<li>Open the report in a browser locally</li>

</ul>
</section>
</section>
<section>
<section id="slide-org9ac8a47">
<h2 id="org9ac8a47">Writing a workflow</h2>
<div class="outline-text-2" id="text-org9ac8a47">
</div>
</section>
</section>
<section>
<section id="slide-org09973e8">
<h3 id="org09973e8">Step 0: Download the data</h3>
<div class="outline-text-3" id="text-org09973e8">
</div>
</section>
<section id="slide-org9b523b5">
<h4 id="org9b523b5">Open <code>workflow/Snakefile</code></h4>
<div class="org-src-container">

<pre  class="src src-snakemake"><code trim><span style="color: #51afef;">rule</span> <span style="color: #c678dd;">bwa_map</span>:
    <span style="color: #ECBE7B;">output</span>:
        <span style="color: #98be65;">"data/samples/A.fastq"</span>
    <span style="color: #ECBE7B;">shell</span>:
        <span style="color: #98be65;">"""</span>
<span style="color: #98be65;">        wget https://github.com/snakemake/snakemake-tutorial-data/archive/v5.4.5.tar.gz</span>
<span style="color: #98be65;">        tar --wildcards -xf v5.4.5.tar.gz --strip 1 "*/data"</span>
<span style="color: #98be65;">        """</span>
</code></pre>
</div>
</section>
</section>
<section>
<section id="slide-org2b63db9">
<h3 id="org2b63db9"><a href="https://snakemake.readthedocs.io/en/stable/tutorial/basics.html#step-1-mapping-reads">Step 1: Mapping reads</a></h3>
<div class="outline-text-3" id="text-org2b63db9">
</div>
</section>
<section id="slide-orgba0fd0a">
<h4 id="orgba0fd0a">Open <code>workflow/Snakefile</code></h4>
<div class="org-src-container">

<pre  class="src src-snakemake"><code trim><span style="color: #51afef;">rule</span> <span style="color: #c678dd;">bwa_map</span>:
    <span style="color: #ECBE7B;">input</span>:
        <span style="color: #98be65;">"/home/test/data/genome.fa"</span>,
        <span style="color: #98be65;">"/home/test/data/samples/A.fastq"</span>
    <span style="color: #ECBE7B;">output</span>:
        <span style="color: #98be65;">"results/mapped_reads/A.bam"</span>
    <span style="color: #ECBE7B;">conda</span>:
        <span style="color: #98be65;">"./envs/bwa.yml"</span>
    <span style="color: #ECBE7B;">shell</span>:
        <span style="color: #98be65;">"bwa mem {input} | samtools view -Sb - &gt; {output}"</span>
</code></pre>
</div>
</section>
<section id="slide-orgfbbd6dc">
<h4 id="orgfbbd6dc">Open <code>workflow/envs/bwa.yml</code></h4>
<div class="org-src-container">

<pre  class="src src-yaml"><code trim><span style="color: #dcaeea;">channels</span>:
  - bioconda
  - conda-forge
  - defaults
<span style="color: #dcaeea;">dependencies</span>:
  - bwa ==0.7.17
  - samtools ==1.9
  - picard ==2.20.1
</code></pre>
</div>
</section>
<section id="slide-org30d4052">
<h4 id="org30d4052">Run</h4>
<div class="org-src-container">

<pre  class="src src-shell"><code trim><span style="color: #ECBE7B;">make</span> dry <span style="color: #dcaeea;">rule</span>=results/mapped_reads/A.bam
<span style="color: #ECBE7B;">make</span> smk <span style="color: #dcaeea;">rule</span>=results/mapped_reads/A.bam
</code></pre>
</div>
</section>
</section>
<section>
<section id="slide-orga3f4e2f">
<h3 id="orga3f4e2f"><a href="https://snakemake.readthedocs.io/en/stable/tutorial/basics.html#step-2-generalizing-the-read-mapping-rule">Step 2: Generalizing the read mapping rule</a></h3>
<div class="outline-text-3" id="text-orga3f4e2f">
</div>
</section>
<section id="slide-org54ee2d1">
<h4 id="org54ee2d1"><code>workflow/Snakefile</code></h4>
<div class="org-src-container">

<pre  class="src src-snakemake"><code trim><span style="color: #51afef;">rule</span> <span style="color: #c678dd;">bwa_map</span>:
    <span style="color: #ECBE7B;">input</span>:
        <span style="color: #98be65;">"data/genome.fa"</span>,
        <span style="color: #98be65;">"data/samples/{sample}.fastq"</span>
    <span style="color: #ECBE7B;">output</span>:
        <span style="color: #98be65;">"results/mapped_reads/{sample}.bam"</span>
    <span style="color: #ECBE7B;">conda</span>:
        <span style="color: #98be65;">"./envs/bwa.yml"</span>
    <span style="color: #ECBE7B;">shell</span>:
        <span style="color: #98be65;">"bwa mem {input} | samtools view -Sb - &gt; {output}"</span>
</code></pre>
</div>
</section>
<section id="slide-orgf3798ac">
<h4 id="orgf3798ac">Run</h4>
<div class="org-src-container">

<pre  class="src src-shell"><code trim><span style="color: #ECBE7B;">make</span> smk <span style="color: #dcaeea;">rule</span>=results/mapped_reads/B.bam
<span style="color: #ECBE7B;">make</span> smk <span style="color: #dcaeea;">rule</span>=<span style="color: #98be65;">"results/mapped_reads/A.bam results/mapped_reads/B.bam"</span>
<span style="color: #ECBE7B;">make</span> smk <span style="color: #dcaeea;">rule</span>=<span style="color: #98be65;">"mapped_reads/{A,B}.bam"</span>
</code></pre>
</div>
</section>
</section>
<section>
<section id="slide-org42589a7">
<h3 id="org42589a7"><a href="https://snakemake.readthedocs.io/en/stable/tutorial/basics.html#step-3-sorting-read-alignments">Step 3: Sorting read alignments</a></h3>
<ul>
<li>Note that Snakemake automatically creates missing directories before jobs are executed.</li>

</ul>
<div class="org-src-container">

<pre  class="src src-snakemake"><code trim><span style="color: #51afef;">rule</span> <span style="color: #c678dd;">samtools_sort</span>:
    <span style="color: #ECBE7B;">input</span>:
        <span style="color: #98be65;">"results/mapped_reads/{sample}.bam"</span>
    <span style="color: #ECBE7B;">output</span>:
        <span style="color: #98be65;">"results/sorted_reads/{sample}.bam"</span>
    <span style="color: #ECBE7B;">conda</span>:
        <span style="color: #98be65;">"envs/sam.yml"</span>
    <span style="color: #ECBE7B;">shell</span>:
        <span style="color: #98be65;">"samtools sort -T results/sorted_reads/{wildcards.sample} "</span>
        <span style="color: #98be65;">"-O bam {input} &gt; {output}"</span>
</code></pre>
</div>
</section>
<section id="slide-org3338a27">
<h4 id="org3338a27">Open <code>workflow/envs/bwa.yml</code></h4>
<div class="org-src-container">

<pre  class="src src-yaml"><code trim><span style="color: #dcaeea;">channels</span>:
  - bioconda
  - conda-forge
  - defaults
<span style="color: #dcaeea;">dependencies</span>:
  - samtools ==1.9
</code></pre>
</div>
</section>
</section>
<section>
<section id="slide-orgdbbdc47">
<h3 id="orgdbbdc47"><a href="https://snakemake.readthedocs.io/en/stable/tutorial/basics.html#step-4-indexing-read-alignments-and-visualizing-the-dag-of-jobs">Step 4: Indexing read alignments and visualizing the DAG of jobs</a></h3>
<div class="outline-text-3" id="text-orgdbbdc47">
</div>
</section>
<section id="slide-orgb234a5b">
<h4 id="orgb234a5b"><code>workflow/Snakefile</code></h4>
<div class="org-src-container">

<pre  class="src src-snakemake"><code trim><span style="color: #51afef;">rule</span> <span style="color: #c678dd;">samtools_index</span>:
    <span style="color: #ECBE7B;">input</span>:
        <span style="color: #98be65;">"results/sorted_reads/{sample}.bam"</span>
    <span style="color: #ECBE7B;">output</span>:
        <span style="color: #98be65;">"results/sorted_reads/{sample}.bam.bai"</span>
    <span style="color: #ECBE7B;">conda</span>:
        <span style="color: #98be65;">"envs/sam.yml"</span>
    <span style="color: #ECBE7B;">shell</span>:
        <span style="color: #98be65;">"samtools index {input}"</span>
</code></pre>
</div>
</section>
<section id="slide-orgfd75723">
<h4 id="orgfd75723">Update all rule</h4>
<p>
<code>workflow/Snakefile</code>
</p>
<div class="org-src-container">

<pre  class="src src-snakemake"><code trim><span style="color: #51afef;">rule</span> <span style="color: #c678dd;">all</span>:
    <span style="color: #ECBE7B;">input</span>:
        <span style="color: #98be65;">"results/sorted_reads/A.bam.bai"</span>,
        <span style="color: #98be65;">"results/sorted_reads/B.bam.bai"</span>,
        <span style="color: #98be65;">"results/sorted_reads/C.bam.bai"</span>,
</code></pre>
</div>
</section>
<section id="slide-org293f460">
<h4 id="org293f460">Run</h4>
<div class="org-src-container">

<pre  class="src src-shell"><code trim><span style="color: #ECBE7B;">make</span> smk
<span style="color: #ECBE7B;">make</span> report
</code></pre>
</div>
<ul>
<li>Download <code>report.html</code></li>

</ul>
</section>
</section>
<section>
<section id="slide-org0a9048a">
<h3 id="org0a9048a"><a href="https://snakemake.readthedocs.io/en/stable/tutorial/basics.html#step-5-calling-genomic-variants">Step 5: Calling genomic variants</a></h3>
<div class="outline-text-3" id="text-org0a9048a">
</div>
</section>
<section id="slide-orgb6d15e0">
<h4 id="orgb6d15e0"><code>expand</code></h4>
<div class="org-src-container">

<pre  class="src src-snakemake"><code trim><span style="color: #c678dd;">expand</span>(<span style="color: #98be65;">"sorted_reads/{sample}.bam"</span>, sample=SAMPLES)
</code></pre>
</div>
<p>
Returns:
</p>
<div class="org-src-container">

<pre  class="src src-python"><code trim>[<span style="color: #83898d;">"sorted_reads/A.bam"</span>, <span style="color: #83898d;">"sorted_reads/B.bam"</span>]
</code></pre>
</div>
</section>
<section id="slide-orgdc1004c">
<h4 id="orgdc1004c">A more complicated expand</h4>
<div class="org-src-container">

<pre  class="src src-snakemake"><code trim><span style="color: #c678dd;">expand</span>(<span style="color: #98be65;">"sorted_reads/{sample}.{replicate}.bam"</span>, sample=SAMPLES, replicate=[<span style="color: #da8548; font-weight: bold;">0</span>, <span style="color: #da8548; font-weight: bold;">1</span>])
</code></pre>
</div>
<p>
Returns:
</p>
<div class="org-src-container">

<pre  class="src src-python"><code trim>[<span style="color: #83898d;">"sorted_reads/A.0.bam"</span>, <span style="color: #83898d;">"sorted_reads/A.1.bam"</span>, <span style="color: #83898d;">"sorted_reads/B.0.bam"</span>, <span style="color: #83898d;">"sorted_reads/B.1.bam"</span>]
</code></pre>
</div>
</section>
<section id="slide-orgc6de378">
<h4 id="orgc6de378">Defining Samples</h4>
<div class="org-src-container">

<pre  class="src src-snakemake"><code trim><span style="color: #dcaeea;">SAMPLES</span> = [<span style="color: #98be65;">"A"</span>, <span style="color: #98be65;">"B"</span>, <span style="color: #98be65;">"C"</span>]
</code></pre>
</div>
</section>
<section id="slide-orge16250c">
<h4 id="orge16250c">Rule for calling variants</h4>
<p>
<code>workflow/Snakefile</code>
</p>
<div class="org-src-container">

<pre  class="src src-snakemake"><code trim><span style="color: #51afef;">rule</span> <span style="color: #c678dd;">bcftools_call</span>:
    <span style="color: #ECBE7B;">input</span>:
        <span style="color: #dcaeea;">fa</span>=<span style="color: #98be65;">"data/genome.fa"</span>,
        <span style="color: #dcaeea;">bam</span>=<span style="color: #c678dd;">expand</span>(<span style="color: #98be65;">"results/sorted_reads/{sample}.bam"</span>, sample=SAMPLES),
        <span style="color: #dcaeea;">bai</span>=<span style="color: #c678dd;">expand</span>(<span style="color: #98be65;">"results/sorted_reads/{sample}.bam.bai"</span>, sample=SAMPLES)
    <span style="color: #ECBE7B;">output</span>:
        <span style="color: #98be65;">"calls/all.vcf"</span>
    <span style="color: #ECBE7B;">conda</span>:
        <span style="color: #98be65;">"../envs/bcf.yml"</span>
    <span style="color: #ECBE7B;">shell</span>:
        <span style="color: #98be65;">"samtools mpileup -g -f {input.fa} {input.bam} | "</span>
        <span style="color: #98be65;">"bcftools call -mv - &gt; {output}"</span>
</code></pre>
</div>
</section>
<section id="slide-org94c00f7">
<h4 id="org94c00f7"><code>workflow/envs/bcf.yml</code></h4>
<div class="org-src-container">

<pre  class="src src-yaml"><code trim><span style="color: #dcaeea;">channels</span>:
  - bioconda
  - conda-forge
  - defaults
<span style="color: #dcaeea;">dependencies</span>:
  - samtools ==1.9
  - bcftools ==1.9
</code></pre>
</div>
</section>
</section>
<section>
<section id="slide-org73e3ef9">
<h3 id="org73e3ef9"><a href="https://snakemake.readthedocs.io/en/stable/tutorial/basics.html#step-6-using-custom-scripts">Step 6: Using custom scripts</a></h3>
<div class="outline-text-3" id="text-org73e3ef9">
</div>
</section>
<section id="slide-org5fda16a">
<h4 id="org5fda16a">In <code>workflow/Snakefile</code></h4>
<div class="org-src-container">

<pre  class="src src-snakemake"><code trim><span style="color: #51afef;">rule</span> <span style="color: #c678dd;">plot_quals</span>:
    <span style="color: #ECBE7B;">input</span>:
        <span style="color: #98be65;">"results/calls/all.vcf"</span>
    <span style="color: #ECBE7B;">output</span>:
        <span style="color: #c678dd;">report</span>(<span style="color: #98be65;">"results/plots/quals.svg"</span>)
    <span style="color: #ECBE7B;">conda</span>:
        <span style="color: #98be65;">"../envs/plot.py"</span>
    <span style="color: #ECBE7B;">script</span>:
        <span style="color: #98be65;">"scripts/plot-quals.py"</span>
</code></pre>
</div>
</section>
<section id="slide-org7f3f0d4">
<h4 id="org7f3f0d4"><code>workflow/scripts/plot-quals.py</code></h4>
<div class="org-src-container">

<pre  class="src src-python"><code trim><span style="color: #51afef;">import</span> matplotlib
matplotlib.use(<span style="color: #98be65;">"Agg"</span>)
<span style="color: #51afef;">import</span> matplotlib.pyplot <span style="color: #51afef;">as</span> plt
<span style="color: #51afef;">from</span> pysam <span style="color: #51afef;">import</span> VariantFile

<span style="color: #dcaeea;">quals</span> = [record.qual <span style="color: #51afef;">for</span> record <span style="color: #51afef;">in</span> VariantFile(snakemake.<span style="color: #c678dd;">input</span>[<span style="color: #da8548; font-weight: bold;">0</span>])]
plt.hist(quals)

plt.savefig(snakemake.output[<span style="color: #da8548; font-weight: bold;">0</span>])
</code></pre>
</div>
</section>
<section id="slide-org7821446">
<h4 id="org7821446"><code>workflow/envs/plot.yml</code></h4>
<div class="org-src-container">

<pre  class="src src-yaml"><code trim><span style="color: #dcaeea;">channels</span>:
  - bioconda
  - conda-forge
  - defaults
<span style="color: #dcaeea;">dependencies</span>:
  - matplotlib
  - pysam
</code></pre>
</div>
</section>
<section id="slide-orgbb51f09">
<h4 id="orgbb51f09">Using this with R</h4>
<ul>
<li><a href="https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#external-scripts">External scripts docs</a></li>
<li>R, Rmd, or ipynb</li>

</ul>
<div class="org-src-container">

<pre  class="src src-R"><code trim><span style="color: #c678dd;">do_something</span> <span style="color: #a9a1e1;">&lt;-</span> <span style="color: #51afef;">function</span>(data_path, out_path, threads, myparam) {
    <span style="color: #5B6268;"># </span><span style="color: #5B6268;">R code</span>
}

do_something(snakemake@input[[<span style="color: #da8548; font-weight: bold;">1</span>]],
             snakemake@output[[<span style="color: #da8548; font-weight: bold;">1</span>]],
             snakemake@threads,
             snakemake@config[[<span style="color: #98be65;">"myparam"</span>]])
</code></pre>
</div>
</section>
</section>
<section>
<section id="slide-orgb9e9ac2">
<h3 id="orgb9e9ac2"><a href="https://snakemake.readthedocs.io/en/stable/tutorial/basics.html#step-7-adding-a-target-rule">Step 7: Adding a target rule</a></h3>
<div class="outline-text-3" id="text-orgb9e9ac2">
</div>
</section>
<section id="slide-orgffbfc3e">
<h4 id="orgffbfc3e"><code>workflow/Snakefile</code></h4>
<div class="org-src-container">

<pre  class="src src-snakemake"><code trim><span style="color: #51afef;">rule</span> <span style="color: #c678dd;">all</span>:
    <span style="color: #ECBE7B;">input</span>:
        <span style="color: #98be65;">"plots/quals.svg"</span>
</code></pre>
</div>
</section>
<section id="slide-org021380b">
<h4 id="org021380b">Run it</h4>
<div class="org-src-container">

<pre  class="src src-shell"><code trim><span style="color: #ECBE7B;">make</span> smk
<span style="color: #ECBE7B;">make</span> report
</code></pre>
</div>
</section>
</section>
<section>
<section id="slide-org16faea4">
<h3 id="org16faea4">Taking it to the Cluster</h3>
<div class="org-src-container">

<pre  class="src src-shell"><code trim>ssh eam150030@ganymede
<span style="color: #ECBE7B;">cd</span> scratch
<span style="color: #ECBE7B;">git</span> clone https://gitlab.com/functional-genomics/analysis/snakemake-tutorial.git
<span style="color: #ECBE7B;">cd</span> snakemake-tutorial
<span style="color: #5B6268;"># </span><span style="color: #5B6268;">pip install cookiecutter</span>
<span style="color: #ECBE7B;">make</span> profile
module add singlarity/2.4.5
<span style="color: #ECBE7B;">make</span> cluster
</code></pre>
</div>
</section>
</section>
<section>
<section id="slide-orgaad127b">
<h2 id="orgaad127b">Nextflow First Look</h2>
<div class="outline-text-2" id="text-orgaad127b">
</div>
</section>
</section>
<section>
<section id="slide-orgdd923a2">
<h3 id="orgdd923a2">Original Goals from the Snakemake template and cookiecutter</h3>
<ul>
<li>Easy to reproduce and run on new datasets</li>
<li>Containerization of software(<code>conda</code>)</li>
<li>Easy to take to the cluster</li>
<li>A template to quickly start a new workflow</li>
<li>Coding best practices(CI pipelines, releases)</li>

</ul>
</section>
</section>
<section>
<section id="slide-orgc139585">
<h3 id="orgc139585"><a href="https://nf-co.re">nf-core pipelines</a></h3>
<ul>
<li>Now that you&rsquo;ve experienced the struggles of creating a pipeline</li>

</ul>
<p>
Running Locally
</p>
<div class="org-src-container">

<pre  class="src src-shell"><code trim>nextflow run nf-core/rnaseq -profile conda <span style="color: #98be65;">\</span>
    --reads <span style="color: #98be65;">'*_R{1,2}.fastq.gz'</span> --genome GRCh37
</code></pre>
</div>
<ul>
<li>If this looks scary see <a href="https://nf-co.re/tools#launch-a-pipeline">nf-core launch</a></li>

</ul>
</section>
</section>
<section>
<section id="slide-orga891d53">
<h3 id="orga891d53">What about this cluster?</h3>
<div class="org-src-container">

<pre  class="src src-shell"><code trim>nextflow run nf-core/rnaseq -profile singularity,utd_ganymede <span style="color: #98be65;">\</span>
    --reads <span style="color: #98be65;">'*_R{1,2}.fastq.gz'</span> --genome GRCh37
</code></pre>
</div>

<p>
<b>Bonus: If you run it on Ganymede it will warn you automagically</b>
</p>
</section>
</section>
<section>
<section id="slide-org6fe263c">
<h3 id="org6fe263c">What about a template?</h3>
<div class="org-src-container">

<pre  class="src src-shell"><code trim>pip install nf-core
nf-core create
</code></pre>
</div>
<ul>
<li><a href="https://nf-co.re/tools">https://nf-co.re/tools</a></li>

</ul>
</section>
</section>
<section>
<section id="slide-orgbf37445">
<h3 id="orgbf37445">Extending workflows</h3>
<ul>
<li>Is possible</li>
<li>However, I&rsquo;d rather feed a one pipeline into another</li>
<li><code>nf-core/rnaseq</code> -&gt; <code>fg/ribo-profile</code></li>

</ul>
</section>
</section>
</div>
</div>
<script src="https://cdn.jsdelivr.net/npm/reveal.js/js/reveal.js"></script>

<script>
// Full list of configuration options available here:
// https://github.com/hakimel/reveal.js#configuration
Reveal.initialize({
multiplex: {
    secret: '', // null if client
    id: '', // id, obtained from socket.io server
    url: '' // Location of socket.io server
},

// Optional libraries used to extend on reveal.js
dependencies: [
 { src: 'https://cdn.jsdelivr.net/npm/reveal.js/lib/js/classList.js', condition: function() { return !document.body.classList; } },
 { src: 'https://cdn.jsdelivr.net/npm/reveal.js/plugin/markdown/marked.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
 { src: 'https://cdn.jsdelivr.net/npm/reveal.js/plugin/markdown/markdown.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
 { src: 'https://cdn.jsdelivr.net/npm/reveal.js/plugin/zoom-js/zoom.js', async: true, condition: function() { return !!document.body.classList; } },
 { src: 'https://cdn.jsdelivr.net/npm/reveal.js/plugin/notes/notes.js', async: true, condition: function() { return !!document.body.classList; } }]
});
</script>
</body>
</html>
